### ISSUE TITLE
Allow to choose any audio device on system

### ISSUE THREAD

**Issue by daemonspudguy** [nicklan/pnmixer#194]
PNMixer is limited specifically to (default) and, on my system, HD Audio Generic. I think the ability to specify in the config file what alsa device it controls would be a good addition.

**Comment by frx-wintermute** [comment]
I agree, I encountered this limitation and initially thought it was a plain bug of PNMixer. I filed a [bug report](https://bugs.debian.org/1142903) on the Debian Bug Tracking System (since I am using the Debian package of PNMixer).
Then I realized it was a limitation: on one of the systems where I tried PNMixer, it could not find the default ALSA device. I found out that this limitation can be somewhat worked around, by explicitly setting the default ALSA device in `~/.asoundrc`.
For instance:
```
$ head -n 3 ~/.asoundrc
# set the default ALSA device
defaults.pcm.!card Generic_1
defaults.ctl.!card Generic_1
```
However, I agree that this limitation is annoying and, possibly, troublesome: in some cases, setting the ALSA device to be controlled by PNMixer as the default ALSA device could have side effects on other programs/libraries. In other words, this workaround is not a general solution.
It would be far better, if PNMixer could be configured to control any ALSA device the user prefers.

**Comment by hasufell** [comment]
Patches welcome. I haven't looked at the code in a decde though 😂
